### PR TITLE
* NEW [rest] add update/read config file APIs.

### PR DESCRIPTION
**Pull request by alvin1221**

### CODE REVIEW

File: nanolib/conf.c [hunk 1]
```diff
@@ -29,6 +29,90 @@ strtrim(char *str)
...
+{
+	char var[20] = { 0 };
+	sprintf(var, "%ld", value);
+	conf_update(fpath, key, var);
+}
+
+void
+conf_update_bool(const char *fpath, const char *key, bool value)
+{
+	char var[20] = { 0 };
```
Comment (lee-emqx) [inline nanolib/conf.c:43]: why 20?

File: nanolib/conf.c [hunk 2]
```diff
@@ -29,6 +29,90 @@ strtrim(char *str)
...
+
+void
+conf_update(const char *fpath, const char *key, char *value)
+{
+	char **linearray = NULL;
+	int    count     = 0;
+	if (value == NULL) {
+		return;
+	}
+	char   descstrlen = strlen(key) + strlen(value) + 2;
```
Comment (lee-emqx) [inline nanolib/conf.c:56]: char?

Comment (alvin1221) [reply nanolib/conf.c:56]: oh, my mistake.

**Review by alvin1221 — COMMENTED** [review]
reviewed

File: nanomq/rest_api.c [hunk 3]
```diff
@@ -30,19 +30,59 @@ static http_msg get_subscriptions(
...
+			rv    = (0);                        \
+		} else {                                    \
+			rv = (-1);                          \
+		}                                           \
+	}
+
+#define getBoolValue(obj, item, key, value, rv)       \
+	{                                             \
+		item = cJSON_GetObjectItem(obj, key); \
+		if (cJSON_IsNumber(item)) {           \
```
Comment (lee-emqx) [inline nanomq/rest_api.c:53]: cJSON_IsBool?

File: nanomq/rest_api.c [hunk 4]
```diff
@@ -30,19 +30,59 @@ static http_msg get_subscriptions(
...
+			rv    = (0);                  \
+		} else {                              \
+			rv = (-1);                    \
+		}                                     \
+	}
+
+#define getStringValue(obj, item, key, value, rv)           \
+	{                                                   \
+		item = cJSON_GetObjectItem(obj, key);       \
+		if (cJSON_IsNumber(item)) {                 \
```
Comment (lee-emqx) [inline nanomq/rest_api.c:64]: cJSON_IsString

File: nanomq/rest_api.c (hunk 3 above)
Comment (alvin1221) [reply nanomq/rest_api.c:53]: Yeah, I should use _cJSON_IsTure()_

**Review by JaylinYu — APPROVED** [review]

**Pull request merged**